scipy/linalg/_solvers.py

```diff
@@ -302,71 +313,149 @@ def solve_discrete_are(a, b, q, r):
     The DARE is defined as
 
     .. math::
-        X = A'XA-(A'XB)(R+B'XB)^-1(B'XA)+Q
+        X = A'XA - (A'XB) (R+B'XB)^{-1} (B'XA) + Q
 
-    It is solved directly using a Schur decomposition method.
+    The limitations for a solution to exist are :
+        
+        * All eigenvalues of ..math:`A` outside the unit disc, should be 
+          controllable. 
 
+        * The associated symplectic pencil (See Notes), should have 
+          eigenvalues sufficiently away from the unit circle. 
+    
     Parameters
     ----------
     a : (M, M) array_like
-        Non-singular, square matrix
+        Square matrix
     b : (M, N) array_like
         Input
     q : (M, M) array_like
         Input
     r : (N, N) array_like
-        Non-singular, square matrix
+        Square matrix
 
     Returns
     -------
     x : ndarray
-        Solution to the continuous Lyapunov equation
+        Solution to the discrete algebraic Riccati equation. If a solution
+        could not be found an empty array is returned.
 
     See Also
     --------
     solve_continuous_are : Solves the continuous algebraic Riccati equation
 
     Notes
     -----
-    Method taken from:
-    Laub, "A Schur Method for Solving Algebraic Riccati Equations."
-    U.S. Energy Research and Development Agency under contract
-    ERDA-E(49-18)-2087.
-    http://dspace.mit.edu/bitstream/handle/1721.1/1301/R-0859-05666488.pdf
+    The equation is solved by forming the extended symplectic matrix pencil, 
+    as described in [1]_, .. math:`H - \lambda J` given by the block matrices
+        
+    
+           [  A   0   B ]             [ I   0   B ] 
+           [ -Q   I   0 ] - \lambda * [ 0  A^T  0 ]
+           [  0   0   R ]             [ 0 -B^T  0 ]
```

Comment:
Needs `::` (two colons) at the end of the preceding text, ie. `... given by the block matrices::`